### DISCUSSION TITLE
iOS 16 Beta AnimationView in UIViewRepresentable won't play

### DISCUSSION THREAD

**Discussion by aharriscrowne** [airbnb/lottie-ios#1667] [General]
## Which Version of Lottie are you using?

Lottie 3.4

## Expected Behavior 

Calling animationView.play(completion:) should play the animation

## Actual Behavior

Calling animationView.play(completion:) dies.  When rendering is .mainThread, it checks for window, but didMoveToWindow is never called.  Not clear if this is iOS 16 intended behavior or a beta bug.

**Comment by calda** [comment]
I'm not familiar with the SwiftUI guarantees around `UIViewRepresentable` and `UIWindow`s. I believe other folks have successfully used the Main Thread rendering engine with SwiftUI and I haven't heard of this issue yet, so I wonder if this is an iOS 16 specific issue.

**Reply by calda** [reply]
Does it work on iOS 15 but specifically not on iOS 16?

**Reply by aharriscrowne** [reply]
It is specifically iOS 16.  I just did a test project and discovered that it is even more specific.  The problem occurs when my UIViewRepresentable (called LottieView) which makes the AnimationView is nested in a SwiftUI Button label.  Like this:

```
var body: some View {
        VStack(spacing: 15) {
            Button {
                selected = true
            } label: {
                if selected {
                    LottieView()
                } else {
                    Text("Push me")
                }
            }

            Text("Hello, world!")
        }
    }
```

This is not specific to a Lottie usage; any UIView you use will not call didMoveToWindow().  I am filing a Apple bug report on it; seems like it is probably not intended behavior.

**Reply by calda** [reply]
Thanks for filing a bug with Apple -- agree that this sounds like a iOS 16 SwiftUI regression

**Reply by ryngonzalez** [reply]
@aharriscrowne Is there a link to the radar submission to follow along?

**Reply by aharriscrowne** [reply]
This now functions correctly in iOS 16.1.  It might also work in 16.0--I have tested in the 16.0 simulator and it works there.
